### PR TITLE
INSERTION SORT IN CPP

### LINKED ISSUE #907: insertion sort in cpp
<hr>

## Description
To show insertion sort in Cpp language via documentation

<hr>

## Domain of Contribution 📊

- [ x] C/CPP
<br>

### PR DESCRIPTION
**Pull request by Ragini-sharma21**
<hr>

## Description 📜

Added a file to show the implementation of insertion sort in CPP
Fixes #907

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by Alankrita9** [comment]
i will add more information  about insertion sort in the code

**Comment by apeksha235** [comment]
Try explaining the code better with comments... Give a detailed description in your own words

File: C_CPP/readme.md [hunk 1]
```diff
@@ -1,45 +1,2 @@
...
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/Algorithmic%20Approaches(Greedy%2C%20Dynamic%2C%20Branch%20%26%20Bound%2C%20NP-completeness%20etc)">Algorithmic Approaches(Greedy, Dynamic, Branch & Bound, NP-completeness etc) </a>
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/OpenGL%20using%20C%2B%2B%20(3D%2C2D%20graphics%20using%20C%2B%2B)">OpenGL using C++ (3D,2D graphics using C++)</a>
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/Unreal%20Engine%20with%20C%2B%2B">Unreal Engine with C++ </a>
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/Juice%20Framework%20with%20C%2B%2B">Juice Framework with C++ </a>
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/C%2B%2B%2014">C++ 14 </a>
-### <a href="https://github.com/girlscript/winter-of-contributing/tree/C_CPP/C_CPP/C%2B%2B%2017">C++ 17 </a>
 
+- [Insertion Sort in Cpp](https://github.com/Ragini-sharma21/winter-of-contributing/blob/milestone/C_CPP/InsertionSort.md]
```
Comment (Suvraneel) [inline C_CPP/readme.md:2]: add to readme of sorting sorting folder

File: C_CPP/InsertionSort.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+## INSERTION SORT
+
```
Comment (Suvraneel) [inline C_CPP/InsertionSort.md:2]: Shift the entire file inside C_CPP/Sorting folder

**Review by Suvraneel — CHANGES_REQUESTED** [review]

File: C_CPP/InsertionSort.md (hunk 2 above)
Comment (Ragini-sharma21) [reply C_CPP/InsertionSort.md]: @Suvraneel sir could you please review the changes.

**Comment by Suvraneel** [comment]
@Ragini-sharma21 Needs more elaboration like @apeksha235  suggested...
Doc is too short (eg, stability could be explained better...)

Also, revert the changes on C_CPP/readme --- !important

File: C_CPP/Data Structures/sorting/InsertionSort.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    cout<<endl;
+}
+```
+- COMPLEXITY OF INSERTION SORT </br>
+
+   Insertion Sort is an easy-to-implement,stable sorting algorithm with time complexity of O(n^2)in the average case and O(n) in the best case when the elements are already sorted and the worst case occurs when the array is sorted in reverse order. For vey small n, Insertion Sort is good for sorting small arrays . It sorts smaller arrays faster than any other sorting algorithm.It is very efficient for semi-sorted arrays . For an array of n elements ,where the maximum divergence an element can have from its "sorted"position is k, insertion sort takes O(n) time.</br>
+   
+   Insertion sort is stable sort with a space complexity of O(1) and it is said to be stable  if two objects with equal keys appear in the same order in sorted output as they appear in the input array to be sorted. Stability is mainly important when we have key value pairs with duplicate keys possible  and we want to sort these objects by keys.</br>
+   A sorting algorithm is stable if-</br>
+   i < j\:\:and\:\:A[i]\equiv A[j]\:\:implies\:\:\pi (i) < \pi (j)</br>
```
Comment (Suvraneel) [inline C_CPP/Data Structures/sorting/InsertionSort.md:51]: Fix mathematical expr

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Pull request merged**